### PR TITLE
Fix/test case errors

### PR DESCRIPTION
**Pull request by justussoh**
This PR includes:
- Fixing of Testcases

### CODE REVIEW

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=84

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-84

File: src/components/QrScanner/QrCamera.tsx [hunk 1]
```diff
@@ -53,7 +53,9 @@ export const QrCamera: FunctionComponent<QrCamera> = ({
   onQrData,
   disabled = false
 }) => {
-  const [hasCameraPermission, setHasCameraPermission] = useState();
+  const [hasCameraPermission, setHasCameraPermission] = useState<
+    boolean | undefined
+  >(undefined);
```
Comment (sebastianquek) [inline src/components/QrScanner/QrCamera.tsx:58]: Do we need to define it as `undefined`? Also, `useState<boolean>` should suffice.

File: src/components/QrScanner/QrCamera.tsx [hunk 2]
```diff
@@ -70,7 +72,7 @@ export const QrCamera: FunctionComponent<QrCamera> = ({
 
   const cameraRef = useRef<Camera>(null);
   const [isCameraReady, setIsCameraReady] = useState(false);
-  const [ratio, setRatio] = useState();
+  const [ratio, setRatio] = useState<string | undefined>();
```
Comment (sebastianquek) [inline src/components/QrScanner/QrCamera.tsx:75]: As above too

File: src/context/db.tsx [hunk 3]
```diff
@@ -8,7 +8,7 @@ import { Database } from "../types";
 
 interface DbContext {
   db?: Database;
-  setDb?: (db: Database) => void;
+  setDb?: React.Dispatch<React.SetStateAction<Database | undefined>>;
```
Comment (sebastianquek) [inline src/context/db.tsx:11]: We typically use the specific modules directly, i.e. `Dispatch<SetStateAction...`

File: src/context/db.tsx [hunk 4]
```diff
@@ -18,14 +18,8 @@ export const DbContext = createContext<DbContext>({
...
 
-export interface DbContextProviderProps {
-  children: object;
-}
-
-export const DbContextProvider: FunctionComponent<DbContextProviderProps> = ({
-  children
-}: DbContextProviderProps) => {
-  const [db, setDb] = useState();
+export const DbContextProvider: FunctionComponent = ({ children }) => {
```
Comment (sebastianquek) [inline src/context/db.tsx:21]: Nice, we don't need that additional interface 👍

File: src/context/config.test.tsx [hunk 5]
```diff
@@ -37,19 +33,16 @@ describe("useConfigContext", () => {
...
-    expect(queryByTestId("printed-network")).toBeNull();
-
-    await waitForElementToBeRemoved(() => queryByTestId("loading-view"));
-    expect(queryByTestId("loading-view")).toBeNull();
-    expect(queryByTestId("printed-network")).not.toBeNull();
-    expect(queryByTestId("printed-network")).toHaveTextContent("mainnet");
+    await wait(() => {
+      expect(queryByTestId("printed-network")).not.toBeNull();
+      expect(queryByTestId("printed-network")).toHaveTextContent("mainnet");
+    });
```
Comment (sebastianquek) [inline src/context/config.test.tsx:45]: Hmm, the existing test tested the idea that the loading view will be shown briefly, what are some ways we can ensure this functionality?

File: src/components/QrScanner/QrCamera.test.tsx [hunk 6]
```diff
@@ -11,7 +11,10 @@ jest.mock("expo-permissions", () => ({
 
 const mockPlatform = (platform: "android" | "ios"): void => {
   jest.resetModules();
-  jest.doMock("Platform", () => ({ OS: platform, select: jest.fn() }));
+  jest.doMock("react-native/Libraries/Utilities/Platform", () => ({
```
Comment (sebastianquek) [inline src/components/QrScanner/QrCamera.test.tsx:14]: Is there a resource that mentions the background of this change?

Comment (justussoh) [reply src/components/QrScanner/QrCamera.test.tsx:14]: https://altany.github.io/react-native/0.61/jest/mocking/upgrade/2020/01/25/mocking-react-native-0.61-modules-with-jest.html

Comment (justussoh) [reply src/components/QrScanner/QrCamera.test.tsx:14]: This was something I found, RN 0.61 removes haste maps

File: src/context/config.test.tsx (hunk 5 above)
Comment (justussoh) [reply src/context/config.test.tsx:45]: Hmm, I realized that we were using useState hooks to show screens, however, useState(boolean) technically only holds 2 states (T/F), undefined is not a valid value. If we want to show a loading view, I would suggest we have an isLoading state as well!

File: src/context/db.tsx (hunk 3 above)
Comment (justussoh) [reply src/context/db.tsx:11]: Okay got it

File: src/components/QrScanner/QrCamera.tsx (hunk 1 above)
Comment (justussoh) [reply src/components/QrScanner/QrCamera.tsx:58]: Hmm im not sure of the default value for useState boolean, need to check if its undefined!

Comment (justussoh) [reply src/components/QrScanner/QrCamera.tsx]: Turns out it returns undefined and is also auto typed if initial value arg is left blank changed the code to reflect the change 👍

File: src/components/QrScanner/QrCamera.tsx (hunk 2 above)
Comment (justussoh) [reply src/components/QrScanner/QrCamera.tsx]: same as above 👍

File: src/context/db.tsx (hunk 3 above)
Comment (justussoh) [reply src/context/db.tsx]: Import from react above 👍

File: src/context/config.test.tsx (hunk 5 above)
Comment (justussoh) [reply src/context/config.test.tsx]: Implemented an isLoaded state to show loading screen state 👍

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=84

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-84

File: src/context/config.tsx [hunk 7]
```diff
@@ -34,13 +34,11 @@ export const useConfigContext = (): ConfigContext =>
...
-  const [config, setConfig] = useState();
+  const [config, setConfig] = useState<Config>(DEFAULT_CONFIG);
+  const [isLoaded, setIsLoaded] = useState<boolean>(false);
 
   const setConfigValue: ConfigContext["setConfigValue"] = (key, value) => {
-    const nextConfig = {
-      ...config,
-      [key]: value
-    };
+    const nextConfig = Object.assign({}, config, { [key]: value });
```
Comment (sebastianquek) [inline src/context/config.tsx:41]: Just noticed this, what's the reason for the change to `.assign`?

File: src/context/config.test.tsx [hunk 8]
```diff
@@ -20,7 +20,9 @@ const TestComponent: FunctionComponent = () => {
   const { config, setConfigValue } = useConfigContext();
   return (
     <View>
-      <Text testID="printed-network">{config.network}</Text>
+      <Text testID="printed-network">
+        {config ? config.network : undefined}
```
Comment (sebastianquek) [inline src/context/config.test.tsx:24]: When will config be falsy?

File: src/context/config.tsx (hunk 7 above)
Comment (justussoh) [reply src/context/config.tsx:41]: `.assign` helps nextConfig inherit type Config, `...` spreader operation changes type to `object` which is not compatible with `setState<Config>`

File: src/context/config.test.tsx (hunk 8 above)
Comment (justussoh) [reply src/context/config.test.tsx:24]: Oh this was from a previous change when config was first unset then upon loaded with be displayed. Should I revert this change?

Comment (sebastianquek) [reply src/context/config.test.tsx:24]: Yep do revert

File: src/context/config.tsx (hunk 7 above)
Comment (sebastianquek) [reply src/context/config.tsx:41]: By using `.assign`, the `nextConfig` type becomes incorrect:
```
const nextConfig: Config & {
    [x: string]: string;
}
```
This accepts any key:value pairs.

Either way, we define `setConfigValue` such that the keys and values are restricted, so we are assured that `nextConfig` is of the correct type. We can also type it by doing the following, but there's no need as it's inferred from the param type of `setConfig`
```
const nextConfig: Config = {
  ...config,
  [key]: value
};
```

File: src/context/config.test.tsx (hunk 8 above)
Comment (justussoh) [reply src/context/config.test.tsx:24]: 👍

File: src/context/config.tsx (hunk 7 above)
Comment (justussoh) [reply src/context/config.tsx:41]: Ahh okay got it 👍

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=84

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-84
